pkg/cnab/provider/install.go

```diff
@@ -3,58 +3,79 @@ package cnabprovider
 import (
 	"fmt"
 
+	"github.com/cnabio/cnab-go/credentials"
+
+	"github.com/hashicorp/go-multierror"
+
 	"github.com/cnabio/cnab-go/action"
 	"github.com/cnabio/cnab-go/claim"
-	"github.com/hashicorp/go-multierror"
 	"github.com/pkg/errors"
-
-	"get.porter.sh/porter/pkg/cnab/extensions"
-	"get.porter.sh/porter/pkg/manifest"
 )
 
-func (d *Runtime) Install(args ActionArguments) error {
-	c, err := claim.New(args.Claim)
-	if err != nil {
-		return errors.Wrap(err, "invalid bundle instance name")
-	}
-
-	b, err := d.LoadBundle(args.BundlePath)
+func (r *Runtime) Install(args ActionArguments) error {
+	b, err := r.ProcessBundle(args.BundlePath)
 	if err != nil {
 		return err
 	}
 
-	err = b.Validate()
+	params, err := r.loadParameters(b, args.Params, claim.ActionInstall)
 	if err != nil {
-		return errors.Wrap(err, "invalid bundle")
+		return errors.Wrap(err, "invalid parameters")
 	}
-	c.Bundle = b
 
-	exts, err := extensions.ProcessRequiredExtensions(b)
+	c, err := claim.New(args.Instanace, claim.ActionInstall, b, params)
 	if err != nil {
-		return errors.Wrap(err, "unable to process required extensions")
+		return errors.Wrap(err, "invalid bundle instance name")
 	}
-	d.Extensions = exts
 
-	params, err := d.loadParameters(c, args.Params, string(manifest.ActionInstall))
+	creds, err := r.loadCredentials(b, args.CredentialIdentifiers)
 	if err != nil {
-		return errors.Wrap(err, "invalid parameters")
+		return errors.Wrap(err, "could not load credentials")
 	}
-	c.Parameters = params
 
-	dvr, err := d.newDriver(args.Driver, c.Installation, args)
+	driver, err := r.newDriver(args.Driver, c.Installation, args)
 	if err != nil {
 		return errors.Wrap(err, "unable to instantiate driver")
 	}
-	i := action.Install{
-		Driver: dvr,
+
+	a := action.New(driver, r.claims)
+	a.SaveAllOutputs = true
+
+	err = a.SaveInitialClaim(c, claim.StatusRunning)
+	if err != nil {
+		return err
 	}
 
-	creds, err := d.loadCredentials(b, args.CredentialIdentifiers)
+	r.printDebugInfo(creds, params)
+
+	opResult, result, err := a.Run(c, creds, r.ApplyConfig(args)...)
 	if err != nil {
-		return errors.Wrap(err, "could not load credentials")
+		err = r.appendFailedResult(err, c)
+		return errors.Wrap(err, "failed to install the bundle")
 	}
 
-	if d.Debug {
+	return a.SaveOperationResult(opResult, c, result)
+}
+
+// appendFailedResult creates a failed result from the operation error and accumulates
+// the error(s).
+func (r *Runtime) appendFailedResult(opErr error, c claim.Claim) error {
```

Comment:
nit: should this function definition live in a more general-purpose file untied to an action?  (`runtime.go` perhaps?)